ConfidentialClientTokenCache/TokenCacheConfidentialClientApplicationExtension.cs

```diff
@@ -0,0 +1,127 @@
+        /// its configuration.
+        /// 
+        /// <code>
+        ///  app.UseTokenCaches(services =>
+        ///  {
+        ///       services.AddDistributedTokenCaches();
+        ///       // Redis token cache
+        ///       // Requires to reference Microsoft.Extensions.Caching.StackExchangeRedis
+        ///       services.AddStackExchangeRedisCache(options =>
+        ///       {
+        ///           options.Configuration = "localhost";
+        ///           options.InstanceName = "Redis";
+        ///       });
+        ///  });
+        /// </code>
+        /// 
+        /// </example>
+        /// <remarks>Don't use this method in ASP.NET Core. Just add use the ConfigureServices method
+        /// instead.</remarks>
+        public static IConfidentialClientApplication UseTokenCaches(
+         this IConfidentialClientApplication app,
+         Action<IServiceCollection> initializeCache)
+        {
+            IHostBuilder hostBuilder = Host.CreateDefaultBuilder()
+              .ConfigureLogging(l => { })
+              .ConfigureServices(services =>
+              {
+                  initializeCache(services);
+              });
+
+            IServiceProvider serviceProvider = hostBuilder.Build().Services;
+            IMsalTokenCacheProvider msalTokenCacheProvider = serviceProvider.GetRequiredService<IMsalTokenCacheProvider>();
+            msalTokenCacheProvider.Initialize(app.UserTokenCache);
+            msalTokenCacheProvider.Initialize(app.AppTokenCache);
+            return app;
+        }
+
+        /// <summary>
+        /// Add an in-memory well partitionned token cache to MSAL.NET confidential client 
+        /// application. Don't use this method in ASP.NET Core: rather use 
+        /// <code>services.AddInMemoryTokenCaches()</code> in ConfigureServices.
+        /// In net462 and net472, you'll need to reference Microsoft.Extensions.Caching.Memory
+        /// </summary>
+        /// <param name="app">Application</param>
+        /// <returns>The application for chaining</returns>
+        public static IConfidentialClientApplication UseInMemoryTokenCaches(
+         this IConfidentialClientApplication app)
+        {
+            // In memory token cache
+            // In net472, requires to reference Microsoft.Extensions.Caching.Memory
+            app.UseTokenCaches(services => 
+            { 
+                services.AddInMemoryTokenCaches();
+            });
+            return app;
+        }
+
+        /// <summary>
+        /// Add a distributed token cache.
+        /// </summary>
+        /// <param name="app">Application</param>
+        /// <param name="initializeDistributedCache">Action taking a <see cref="IServiceCollection"/>
+        /// and by which you initialize your distributed cache</param>
+        /// <returns>The application for chaining</returns>
+        public static IConfidentialClientApplication UseDistributedTokenCache(
```

Comment:
I left this method for the moment, but I don't think it improves the developer experience. See the discussion in the PR.